pyiceberg/table/update/snapshot.py

```diff
@@ -239,7 +241,21 @@ def _summary(self, snapshot_properties: Dict[str, str] = EMPTY_DICT) -> Summary:
             truncate_full_table=self._operation == Operation.OVERWRITE,
         )
 
+    def refresh(self) -> TableMetadata:
+        try:
+            table = self._transaction._table.refresh()
+            return table.metadata
+        except Exception:
+            return self._transaction._table.metadata
```

Comment:
Any specific reason to swallow this exception? If there are any errors, I think it is likely that the commit will fail as well.